### PR TITLE
CM-628: The AddEndpointsApiExplorer Method in ASP.NET Core

### PR DESCRIPTION
**Pull request by aineaTheSecond**

### CODE REVIEW

File: aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Controllers/WeatherForecastController.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{
+
+    [ApiController]
+    [Route("[controller]")]
+    public class WeatherForecastController : ControllerBase
+    {
+        private static readonly string[] Summaries = new[]
+        {
+        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
+    };
```
Comment (karimkod) [inline aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Controllers/WeatherForecastController.cs:13]: formatting and break the long line to multiple lines.

File: aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Program.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+app.UseHttpsRedirection();
+
+app.UseAuthorization();
+
+app.MapControllers();
+
+app.Run();
+
+public partial class Program { }
```
Comment (karimkod) [inline aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Program.cs:30]: Why do we need this?

File: aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Program.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+var app = builder.Build();
+
+app.MapGet("/", () => "A test endpoint");
```
Comment (karimkod) [inline aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Program.cs:13]: It's a good practice to declare API after all the middlewares are declared (UseXXX) , I suggest  we move this after app.UseAuthorization();

**Review by karimkod — CHANGES_REQUESTED** [review]

File: aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Program.cs (hunk 3 above)
Comment (aineaTheSecond) [reply aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Program.cs]: Sure, this is done.

File: aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Program.cs (hunk 2 above)
Comment (aineaTheSecond) [reply aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Program.cs:30]: I added it in order to get visibility of the minimal API endpoint for testing.

File: aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Controllers/WeatherForecastController.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            "Balmy",
+            "Hot",
+            "Sweltering",
+            "Scorching"
+        };
+
+
+        public WeatherForecastController()
+        {
+        }
```
Comment (karimkod) [inline aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Controllers/WeatherForecastController.cs:26]: This constructor can be removed

**Review by karimkod — CHANGES_REQUESTED** [review]

File: aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Program.cs (hunk 2 above)
Comment (karimkod) [reply aspnetcore-webapi/AddEndpointsApiExplorerMethod/AddEndpointsApiExplorerMethod/Program.cs]: All right, thanks for the explanation.

**Pull request merged**